### PR TITLE
Hide previous question attempts

### PR DESCRIPTION
**Pull request by chrisjpurdy**
**This change requires the following addition to the segue config:** `DISPLAY_SETTING=...,HIDE_QUESTION_ATTEMPTS`.

---

This PR allows users to opt-in to having all previous best attempts to questions hidden if and when they revisit them.

I noticed a page crash when testing this locally - I can't reproduce it or work out what the issue might be (the code seems solid), so unhandled race conditions might be something to look out for (I couldn't find any myself).

Message displayed if best attempt has been hidden, only displayed if there is no current attempt: (the 2nd-to-last commit adds "Your" to the beginning of this message)
image

Beta feature tab on user account, with the checkbox to enable this feature:
image
The tooltip reads: "This feature is helpful for revision for example - you can attempt all of the questions on the platform again, without seeing your previous answers (correct or not)."

### CODE REVIEW

**Review by Morpheu5 — APPROVED** [review]
Looks good aside from that one comment.

File: src/IsaacApiTypes.tsx [hunk 1]
```diff
@@ -458,9 +458,11 @@ export interface QuantityDTO extends ChoiceDTO {
     units?: string;
 }
 
+export type BestAttemptHidden = null;
+export const BEST_ATTEMPT_HIDDEN: BestAttemptHidden = null;
 export interface QuestionDTO extends ContentDTO {
     hints?: ContentBaseDTO[];
-    bestAttempt?: QuestionValidationResponseDTO;
+    bestAttempt?: QuestionValidationResponseDTO | BestAttemptHidden;
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/IsaacApiTypes.tsx:465]: The extra type is needed so we can conditionally show "Your previous attempt has been hidden". If we just used `null` or `undefined` for this, we wouldn't be able to tell the different between no current attempt and a hidden current attempt in order to display the message.

[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/IsaacApiTypes.tsx:465]: Actually to be fair, this code doesn't really fully achieve that goal anyway: it should be `BEST_ATTEMPT_HIDDEN = "BEST_ATTEMPT_HIDDEN"` in order to make the distinction at runtime too. Giving `null` any meaning other than "non-existent" feels like an antipattern

[the remark this answers is not in the document]
Comment (Morpheu5) [reply src/IsaacApiTypes.tsx:465]: Yeah, I'm not sure that, once it's transpiled down to JS, the engine would be able to tell the difference between a `null | undefined` and something that was originally aliased to `null` but is now just `null`.

**Review by jsharkey13 — APPROVED** [review]

**Comment by jsharkey13** [comment]
I'd like those wording changes made; I'm interested in opinions on message-at-top vs message-inline but have no real preference and so once the types thing is sorted and the wording updated I think this is good to go.

**Pull request merged**